migration/v3/model/bucket.go

```diff
@@ -0,0 +1,56 @@
+package model
+
+import (
+	"github.com/google/uuid"
+	liberr "github.com/konveyor/controller/pkg/error"
+	"gorm.io/gorm"
+	"os"
+	"path"
+	"time"
+)
+
+type Bucket struct {
+	Model
+	Path       string `gorm:"<-:create;uniqueIndex"`
+	Expiration *time.Time
+}
+
+func (m *Bucket) BeforeCreate(db *gorm.DB) (err error) {
+	if m.Path == "" {
+		uid := uuid.New()
+		m.Path = path.Join(
+			Settings.Hub.Bucket.Path,
+			uid.String())
+		err = os.MkdirAll(m.Path, 0777)
+		if err != nil {
+			err = liberr.Wrap(
+				err,
+				"path",
+				m.Path)
+		}
+	}
+	return
+}
+
+type BucketOwner struct {
+	BucketID *uint `gorm:"index" ref:"bucket"`
+	Bucket   *Bucket
+}
+
+func (m *BucketOwner) BeforeCreate(db *gorm.DB) (err error) {
+	if !m.HasBucket() {
```

Comment:
**hint**: The bucket is created only when not already set by the creator.  This is to support:
- migration
- Task groups which creates it's tasks with a shared bucket.